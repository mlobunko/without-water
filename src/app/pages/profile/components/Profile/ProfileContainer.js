import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as actions from '../../actions';
import { getCity } from '../../selectors';
import ProfileComponent from './ProfileComponent';

import error from 'shared/error';
import fetching from 'shared/fetching';

export class ProfileContainer extends React.Component {
  componentDidMount = () => {
    this.props.errorActionsSetNull();
    if (!this.props.city) {
      this.props.fetchProfile();
    }
  };
  render() {
    return (
      <ProfileComponent
        city={this.props.city}
        fetching={this.props.fetching}
        isError={this.props.errorState.isError}
        languages={this.props.languages}
        serverError={this.props.errorState.serverError}
        loginError={this.props.errorState.loginError}
      />
    );
  }
}

const mapStateToProps = state => ({
  city: getCity(state),
  fetching: fetching.selectors.getAll(state),
  errorState: error.selectors.getAll(state)
});

const madDispatchToProps = {
  fetchProfile: actions.fetch,
  errorActionsSetNull: error.actions.setNull
};

export default connect(
  mapStateToProps,
  madDispatchToProps
)(ProfileContainer);

ProfileContainer.propTypes = {
  city: PropTypes.string.isRequired,
  fetching: PropTypes.bool.isRequired,
  errorState: PropTypes.shape({
    isError: PropTypes.bool.isRequired,
    loginError: PropTypes.string,
    serverError: PropTypes.string
  })
};
