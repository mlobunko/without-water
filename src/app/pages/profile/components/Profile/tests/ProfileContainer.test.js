import React from 'react';
import { shallow } from 'enzyme';

import { ProfileContainer } from '../ProfileContainer';

const initialProps = {
  fetchProfile: jest.fn(),
  errorActionsSetNull: jest.fn(),
  city: 'London',
  fetching: true,
  errorState: {
    isError: true,
    loginError: 'error',
    serverError: 'error'
  }
};

describe('handlers', () => {
  describe('componentDidMount', () => {
    describe('not fetch', () => {
      const props = {
        ...initialProps
      };

      beforeEach(() => {
        jest.resetAllMocks();
        shallow(<ProfileContainer {...props} />);
      });

      it('call error to null', () => {
        expect(props.errorActionsSetNull).toHaveBeenCalledTimes(1);
      });

      it('fetch is not called', () => {
        expect(props.fetchProfile).toHaveBeenCalledTimes(0);
      });
    });

    describe('fetch', () => {
      const props = {
        ...initialProps,
        city: ''
      };
      beforeEach(() => {
        jest.resetAllMocks();
        shallow(<ProfileContainer {...props} />);
      });

      it('call error to null', () => {
        expect(props.errorActionsSetNull).toHaveBeenCalledTimes(1);
      });
      it('call fetch', () => {
        expect(props.fetchProfile).toHaveBeenCalledTimes(1);
      });
    });
  });
});

describe('render', () => {
  const props = {
    ...initialProps
  };
  it('renders correctly', () => {
    const wrapper = shallow(<ProfileContainer {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
