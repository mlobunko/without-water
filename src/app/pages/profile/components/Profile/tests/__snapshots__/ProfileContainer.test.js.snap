// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<ProfileComponent
  city="London"
  fetching={true}
  isError={true}
  loginError="error"
  serverError="error"
/>
`;
